modules/aws/eks/iam.tf

```diff
@@ -0,0 +1,125 @@
+data "aws_iam_policy_document" "cert_manager" {
+  statement {
+    sid    = "AllowRoute53Change"
+    effect = "Allow"
+    actions = [
+      "route53:GetChange"
+    ]
+    resources = ["arn:aws:route53:::change/*"]
+  }
+  statement {
+    sid    = "AllowRoute53Record"
+    effect = "Allow"
+    actions = [
+      "route53:ChangeResourceRecordSets",
+      "route53:ListResourceRecordSets"
+    ]
+    resources = ["arn:aws:route53:::hostedzone/*"]
+  }
+  statement {
+    sid    = "AllowRoute53List"
+    effect = "Allow"
+    actions = [
+      "route53:ListHostedZonesByName"
+    ]
+    resources = ["*"]
+  }
+}
+
+module "cert_manager" {
+  source = "../irsa"
+
+  name = "cert-manager-${var.name}${var.eks_name_suffix}"
+  oidc_providers = [
+    {
+      url = aws_iam_openid_connect_provider.this.url
+      arn = aws_iam_openid_connect_provider.this.arn
+    }
+  ]
+  kubernetes_namespace       = "cert-manager"
+  kubernetes_service_account = "cert-manager"
+  policy_json                = data.aws_iam_policy_document.cert_manager.json
+}
+
+data "aws_iam_policy_document" "external_dns" {
+  statement {
+    sid    = "AllowRoute53Change"
+    effect = "Allow"
+    actions = [
+      "route53:ChangeResourceRecordSets"
+    ]
+    resources = ["arn:aws:route53:::hostedzone/*"]
+  }
+  statement {
+    sid    = "AllowRoute53List"
+    effect = "Allow"
+    actions = [
+      "route53:ListHostedZones",
+      "route53:ListResourceRecordSets"
+    ]
+    resources = ["*"]
+  }
+}
+
+module "external_dns" {
+  source = "../irsa"
+
+  name = "external-dns-${var.name}${var.eks_name_suffix}"
+  oidc_providers = [
```

Comment:
Not a requirement, but wouldn't it be nice to make this an array of maps? In that way, we would be able to see that it belongs to eks or eks1 or eks2 as an example.